pkg/plugin/plugin.go

```diff
@@ -78,6 +82,55 @@ type Info struct {
 	ConfigAttributes json.RawMessage `db:"config_attrs" json:"config_attrs"` // ConfigOption(s) as json-encoded list
 }
 
+// NewInfo constructs a new Info to be used in Plugin.GetInfo.
+//
+// The configOptions will be enriched with defaults from the typePtr, being a pointer to the Plugin implementation.
+func NewInfo(typePtr interface{}, name, version, author string, configOptions []*ConfigOption) (*Info, error) {
+	err := defaults.Set(typePtr)
+	if err != nil {
+		return nil, err
+	}
+
+	typeDefaults := make(map[string]any)
+	typeRefT := reflect.TypeOf(typePtr).Elem()
+	typeRefV := reflect.ValueOf(typePtr)
+	for i := 0; i < typeRefT.NumField(); i++ {
+		if !typeRefT.Field(i).IsExported() {
```

Comment:
I would simply cache the result of `typeRefT.Field(i)` here instead of repeatedly invoking it.